cli/flags.rs

```diff
@@ -2420,17 +2429,13 @@ fn watch_arg_parse(
   matches: &clap::ArgMatches,
   allow_extra: bool,
 ) {
-  flags.watch = match matches.values_of("watch") {
-    Some(f) => {
-      let extra_files = f.map(PathBuf::from).collect();
-      if allow_extra {
-        Some(extra_files)
-      } else {
-        Some(vec![])
-      }
+  if allow_extra {
+    if let Some(f) = matches.values_of("watch") {
+      flags.watch = Some(f.map(PathBuf::from).collect());
```

Comment:
Will this still work with `deno run --watch`? Is there a unit test that covers this case?

Reply:
Actually I see no unit test that covers `deno run --watch=file1,file2`, could you add it @jespertheend?

Reply:
https://github.com/denoland/deno/blob/6c324acf2363e88293ab94cf3de6c9d7a264b55d/cli/tests/integration/watcher_tests.rs#L734 should cover `deno run --watch`. But now that I think about it, it seems like this shouldn't work. I'll investigate.
I'll also add the extra test 👍

Reply:
Ah this works because `matches.values_of` returns an empty vec when used as `--watch` without any files.
I've added the extra unit test.